logstash-core/src/main/java/org/logstash/FieldReference.java

```diff
@@ -139,8 +139,11 @@ private static int calculateHash(final String key, final String[] path, final in
     }
 
     private static FieldReference parseToCache(final CharSequence reference) {
```

Comment:
nitpick: method name is no longer ALWAYS accurately descriptive; if we brought this decision up to `FieldReference#from(CharSequence)`, we could maintain a little more clarity.

~~~ java
     public static FieldReference from(final CharSequence reference) {
         // atomicity between the get and put is not important
         final FieldReference result = CACHE.get(reference);
         if (result != null) {
             return result;
         }
         if (CACHE.size() <= 10_000) {
             return parseToCache(reference);
         }
         return parse(reference);
     }
~~~

Which simplifies this method to:

~~~ java
    private static FieldReference parseToCache(final CharSequence reference) {
        final FieldReference result = deduplicate(parse(reference));
        CACHE.put(reference, result);
        return result;
    }
~~~

Reply:
@yaauie I'd rather not do that, it increases the compiled size of `from` in the default use case while if we keep the check outside the conditional in the other method, the use case that doesn't exceed 10k cache size will eventually eliminate that code and inline better upstream right?